CalibPPS/ESProducers/plugins/CTPPSInterpolatedOpticalFunctionsESSource.cc

```diff
@@ -28,19 +29,26 @@ class CTPPSInterpolatedOpticalFunctionsESSource : public edm::ESProducer {
 private:
   edm::ESGetToken<LHCOpticalFunctionsSetCollection, CTPPSOpticsRcd> opticsToken_;
   edm::ESGetToken<LHCInfo, LHCInfoRcd> lhcInfoToken_;
+  edm::ESGetToken<LHCInfoPerLS, LHCInfoPerLSRcd> lhcInfoPerLSToken_;
+  edm::ESGetToken<LHCInfoPerFill, LHCInfoPerFillRcd> lhcInfoPerFillToken_;
   std::shared_ptr<LHCInterpolatedOpticalFunctionsSetCollection> currentData_;
   float currentCrossingAngle_;
   bool currentDataValid_;
+  bool useNewLHCInfo_;
 };
 
 //----------------------------------------------------------------------------------------------------
 //----------------------------------------------------------------------------------------------------
 
 CTPPSInterpolatedOpticalFunctionsESSource::CTPPSInterpolatedOpticalFunctionsESSource(const edm::ParameterSet &iConfig)
-    : currentCrossingAngle_(-1.), currentDataValid_(false) {
+    : currentCrossingAngle_(-1.),
+      currentDataValid_(false),
+      useNewLHCInfo_(iConfig.getParameter<bool>("useNewLHCInfo")) {
   auto cc = setWhatProduced(this, iConfig.getParameter<std::string>("opticsLabel"));
   opticsToken_ = cc.consumes(edm::ESInputTag("", iConfig.getParameter<std::string>("opticsLabel")));
   lhcInfoToken_ = cc.consumes(edm::ESInputTag("", iConfig.getParameter<std::string>("lhcInfoLabel")));
+  lhcInfoPerLSToken_ = cc.consumes(edm::ESInputTag("", iConfig.getParameter<std::string>("lhcInfoPerLSLabel")));
+  lhcInfoPerFillToken_ = cc.consumes(edm::ESInputTag("", iConfig.getParameter<std::string>("lhcInfoPerFillLabel")));
```

Comment:
I think these tokens could be moved to the init list, and have them `const`-s, is that not the case?

Reply:
I don't think that's possible, when the `ConsumesCollector` is an object instantiated from within the class constructor ([dxr](https://cmssdt.cern.ch/dxr/CMSSW/search?q=cc.consumes&case=true&offset=127)).